src/main/java/org/skriptlang/reflect/syntax/event/BukkitCustomEvent.java

```diff
@@ -23,7 +23,8 @@ public class BukkitCustomEvent extends Event implements Cancellable {
   private final Map<ClassInfo<?>, Object> eventValueMap;
   private final Map<String, Object> dataMap;
   private boolean isCancelled = false;
-
+  private boolean isAsynchronousAllowed = false;
+  
   public BukkitCustomEvent(String name) {
     this(name, !Bukkit.isPrimaryThread());
```

Comment:
it seems like it would be better to instead record whether this event is asynchronous and pass that here instead of basing it on whether we're on the main thread

Reply:
Bukkit won’t allow us to trigger an asynchronous event from the main thread, and a synchronous one from a non-main thread, is it worth shifting the responsibility for this to the end user?